service/awsconfig/v10/resource/cloudformation/update.go

```diff
@@ -206,18 +222,11 @@ func shouldScale(currentState, desiredState StackState) bool {
 // shouldUpdate determines whether the reconciled guest cluster should be
 // updated. A guest cluster is only allowed to update in the following cases.
 //
-//     The worker count changes (indicates scaling).
-//     The version bundle version changes (indicates updates).
 //     The instance type of master nodes changes (indicates updates).
 //     The instance type of worker nodes changes (indicates updates).
+//     The version bundle version changes (indicates updates).
 //
 func shouldUpdate(currentState, desiredState StackState) bool {
-	// Check scaling related properties.
-	if currentState.WorkerCount != desiredState.WorkerCount {
-		return true
-	}
-
-	// Check updates related properties.
```

Comment:
This is super confusing. We did this initially to be smart and scale also on updates. This does not seem to work out now because the conditions screw the code path. So we have to be more strict and update when we want to update and scale when we want to scale. This is now due to the master shutdown magic. I figured that there is also no real world example that exists which would benefit from the "smart" condition we had here. This now just fixes scaling and all should be fine again.

Reply:
Makes sense to split scaling and updates. IMO the most important thing is for updates to be reliable. We can optimise later once we've learned more about the update process.